hack/check-latest-images.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash
+
+# Copyright The Shipwright Contributors
+#
+# SPDX-License-Identifier: Apache-2.0
+
+# Check the latest image and replace
+
+# Usage
+# ./check-latest-image.sh <FILE> <IMAGE> <IMAGE_URL>
+
+set -euo pipefail
+
+# Check the number of arguments passed and exit if less than three
+if [ $# -lt 3 ]; then
+        echo "USAGE: $0 <FILE> <IMAGE> <IMAGE_URL>"
+        exit 1
+fi
+
+# Preserve the literal value of . and /
+# sed -e 's/[]\/$*.^[]/\\&/g'
+
+# Set env for image, current version and lastest version
+IMAGE=$(printf '%s\n' "$(echo ${1})" | sed -e 's/[]\/$*.^[]/\\&/g')
+CURRENT_TAG=$(printf '%s\n' "$(grep -r -h -m 1 -o ${1}'[^ ]*' ${3} | head -n 1 | cut -d':' -f2)" | sed -e 's/[]\/$*.^[]/\\&/g')
+LATEST_TAG=$(printf '%s\n' "$(until curl -s ${2} | jq -r '.name'; do sleep 1; done)" | sed -e 's/[]\/$*.^[]/\\&/g')
+if echo ${IMAGE} | grep trivy 2>&1>/dev/null; then
+        LATEST_TAG=$(printf '%s\n' "$(until curl -s ${2} | jq -r '.name'; do sleep 1; done)" | sed -e 's/[]\/$*.^[]/\\&/g')
+        LATEST_TAG="${LATEST_TAG:1}"
+fi
+if echo ${IMAGE} | grep buildah 2>&1>/dev/null; then
+        LATEST_TAG=$(printf '%s\n' "$(until curl -s ${2} | jq -r '.tags | .[0].name'; do sleep 1; done)" | sed -e 's/[]\/$*.^[]/\\&/g')
+fi
+
+# Directory to be used for search
+DIRECTORY=${3}
```

Comment:
As a best practice, I suggest that you move this line to the top after your count of inputs check, and use named variables for all inputs.

```suggestion
DIRECTORY="${3}"
```